pkg/distributor/distributor.go

```diff
@@ -1502,16 +1518,13 @@ func (d *Distributor) logLabelValueTooLongSummaries(userID string, valueTooLongS
 
 // prePushMaxSeriesLimitMiddleware enforces the per-tenant max series limit when the usage-tracker service is enabled.
 func (d *Distributor) prePushMaxSeriesLimitMiddleware(next PushFunc) PushFunc {
-	return func(ctx context.Context, pushReq *Request) error {
+	return WithCleanup(next, func(next PushFunc, ctx context.Context, pushReq *Request) error {
 		// If the usage-tracker client hasn't been created it means usage-tracker is disabled
 		// for this instance.
 		if d.usageTrackerClient == nil {
```

Comment:
nitpick: We could just not add `prePushMaxSeriesLimitMiddleware` to the middleware chain if `d.usageTrackerClient == nil`, then the middleware can assume `d.usageTrackerClient` is set.